### PR TITLE
password reset modals

### PR DESCRIPTION
**Pull request by kara-o**
## This PR contains the code for client-side password reset flow.

- User clicks on 'Forgot Password' link, and is prompted to enter email (server will always return 202 here, for security)

image

- If valid user, token has been sent to their email

image
image

- If valid token (matches and still within time frame), user can enter new password

image

- Once successfully updated, user is alerted in modal and via email

image
image

### Additional information and considerations

- Async Storage was used to persist state only for stage when user is asked for token.  Once user successfully resets password, logs in, or presses 'back' and returns to email input stage, the stage is reset back to initial.  The new password entry stage will not persist, they will be back to token entry if app is refreshed.

- Currently the only password validations on front end are: check if both password and password confirmation match, and check that password is at least 8 chars long.

- **For testing out email, please message me for SendGrid credentials to add to .env file**

### CODE REVIEW

File: src/state/actions/passwordReset.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	setError: (value: string) => void;
+}
+
+export type RequestResetTokenProps = ResetPasswordPropsBase;
+
+export const requestResetToken = async (store, {
+	onComplete, input, setIsSubmitting, setError,
+}: RequestResetTokenProps) => {
+	const { userIdentity } = store.state;
+	const endpoint = `http://localhost:3000/password_resets/${userIdentity}/`;
```
Comment (emarti43) [inline src/state/actions/passwordReset.ts]: You can use the  `API_BASE_URL` env variable we have to handle different environments (loaded from `environment.ts`).

**Comment by emarti43** [comment]
I got it to work on my end. It seems to be working pretty well 👍. I'm wondering if an option to cancel the reset (clearing asyncStorage) could be implemented later (Not in this PR, as there's a lot going on here)

File: src/state/actions/passwordReset.ts (hunk 1 above)
Comment (kara-o) [reply src/state/actions/passwordReset.ts]: Fixed! Thanks :)

**Comment by kara-o** [comment]
as it is currently, when the user clicks the back button and goes back to the email entry form, the async storage is cleared.  it is also cleared when user logs in successfully (like if they remember their password), and also when the password is reset.

**Review by lofiandcode — DISMISSED** [review]
Everything is working well on my local machine. Great job Kara!

**Review by lofiandcode — APPROVED** [review]

**Pull request merged**